geth/common/types.go

```diff
@@ -177,6 +177,8 @@ type JailCell interface {
 	Set(string, interface{}) error
 	// Get a value from VM.
 	Get(string) (otto.Value, error)
+	// GetObjectValue returns the given name's otto.Value from the given otto.Value v. Should only be needed in tests.
+	GetObjectValue(otto.Value, string) (otto.Value, error)
```

Comment:
Hmm, it looks like otto.Value isn't thread-safe because it calls underlying runtime somewhere.
I wonder if we should return it at all (should `Get` be public).
Do we use it somewhere in our code?
Maybe we can make a wrapper that makes the calls safe?

Reply:
@mandrigin Looks like `Get` is only used in `geth/jail/internal` and in tests:

![image](https://user-images.githubusercontent.com/138074/36438253-a1bcc5c2-1669-11e8-9a16-f60b1b25ea6d.png)

> Maybe we can make a wrapper that makes the calls safe?

Do you mean making `Get` return a wrapper to otto.Value that is thread-safe? It would be the cleanest solution, but probably overkill for something that'll end up only being used by tests.

Reply:
If `Get` is used only internally in the package, it shouldn't be a public method then. That would also help us.

We can have a very simple wrapper that contains something like `DoSafely(f func (otto.Value))` that does
```
func (w *ValueWrapper) DoSafely(f func (otto.Value) {
w.mu.Lock()
defer w.mu.Unlock()
f(w.value)
}
```

That might be enough for internal use.

But in general, `Get` shouldn't be public and shouldn't exist in that interface at all.

Reply:
Makes sense. I was initially going to do a generic method like that, but wasn't aware that `Get` shouldn't be public to start with. I'll make that change then.

Reply:
Hmm, looks like building a wrapper isn't as viable as it looked at first sight, when you look at code like `TestCellFetchErrorRace()` in `cell_test.go`: https://github.com/status-im/status-go/blob/fix/test-race-condition/geth/jail/cell_test.go#L163

The Javascript call to `__captureSuccess` will produce an `otto.Value` instance internally a, for which we have no wrapper, so we'd also need to create a `Cell.WrapValue` method to use in https://github.com/status-im/status-go/blob/fix/test-race-condition/geth/jail/cell_test.go#L146.

Reply:
Yeah, that isn't easy... Is `Set` also only used in tests?

Reply:
Huh, this appears to be the only usage actually.

Reply:
Yeah, or we can remove these methods from `common/types#JailCell`, and make tests use  `*Cell`, where all our production code will use `*Cell`.

Alternatively, we can just make `*vm.VM` as a private field instead of a part of the composition to hide all unwanted methods.